app/models/text_component.rb

```diff
@@ -9,6 +9,8 @@ class TextComponent < ActiveRecord::Base
 
   accepts_nested_attributes_for :triggers
 
+  before_create :assign_default_channel
```

Comment:
I changed the before_validation to expect channels. This breaks ~50 tests of different kinds. Changing the default text_component factory will only fix part of them.

Can you have a look?